libraries/chain/block_state.cpp

```diff
@@ -297,6 +303,16 @@ digest_type block_state::get_finality_mroot_claim(const qc_claim_t& qc_claim) co
    return get_validation_mroot(next_core_metadata.final_on_strong_qc_block_num);
 }
 
+finality_data_t block_state::get_finality_data() const {
+   finality_data_t finality_data {
+      // other fields take the default values set by finality_data_t definition
+      .action_mroot = action_mroot,
+      .base_digest  = compute_base_digest()  // from block_header_state
```

Comment:
since `base_digest` is cached, we should use the cached value instead of recomputing it here.

Reply:
Thanks for this great catch!